### DISCUSSION TITLE
Git Branching Strategy

### DISCUSSION THREAD

**Discussion by nkoshell** [open-metadata/OpenMetadata#16687] [General]
Minor release branches are started from the main branch, but are not merged back when the release is finished.

- How do you ensure backward compatibility between minor versions?
- Do all changes from the previous minor version carry over to the current minor version?
- Could you describe your approach to product development and branching strategy in more detail?

**Comment by harshach** [comment]
1. All code changes lands in main
2. Release branch is made from main
3. If its minor release, a change is cherry-picked from main
4. Major release branches gets rebased from main
